src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -283,23 +284,64 @@ void Ekf::predictState(const imuSample &imu_delayed)
 	_height_rate_lpf = _height_rate_lpf * (1.0f - alpha_height_rate_lpf) + _state.vel(2) * alpha_height_rate_lpf;
 }
 
-void Ekf::resetGlobalPosToExternalObservation(double lat_deg, double lon_deg, float accuracy, uint64_t timestamp_observation)
+void Ekf::resetGlobalPosToExternalObservation(double lat_deg, double lon_deg, float accuracy,
+		uint64_t timestamp_observation)
 {
+
 	if (!_pos_ref.isInitialized()) {
+		ECL_INFO("Global position reference not initialized, set home position first");
 		return;
 	}
 
 	// apply a first order correction using velocity at the delated time horizon and the delta time
 	timestamp_observation = math::min(_time_latest_us, timestamp_observation);
 	const float dt = _time_delayed_us > timestamp_observation ? static_cast<float>(_time_delayed_us - timestamp_observation)
 			 * 1e-6f : -static_cast<float>(timestamp_observation - _time_delayed_us) * 1e-6f;
-
 	Vector2f pos_corrected = _pos_ref.project(lat_deg, lon_deg) + _state.vel.xy() * dt;
 
-	resetHorizontalPositionTo(pos_corrected, sq(math::max(accuracy, 0.01f)));
+	if (!_control_status.flags.in_air || accuracy < 1.f) {
```

Comment:
It wouldn't hurt to check the accuracy is > 0 here.

Reply:
if the accuracy is < 0, `math::max(accuracy, FLT_EPSILON)` will handle it, or what do you mean?

Reply:
As a way to verify that the good accuracy (accuracy < 1.f) is real vs slipping through uninitialized or zero'd.